Xwt/Xwt/Calendar.cs

```diff
@@ -0,0 +1,153 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+using System;
+using Xwt.Backends;
+using System.ComponentModel;
+
+namespace Xwt
+{
+	[BackendType (typeof(ICalendarBackend))]
+	public class Calendar : Widget
+	{
+		EventHandler valueChanged, doubleClick;
+
+		static Calendar ()
+		{
+			MapEvent (CalendarEvent.ValueChanged, typeof(Calendar), "OnValueChanged");
+			MapEvent (CalendarEvent.DoubleClick, typeof(Calendar), "OnDoubleClick");
+		}
+
+		protected new class WidgetBackendHost: Widget.WidgetBackendHost, ICalendarEventSink
+		{
+			public void OnValueChanged ()
+			{
+				((Calendar)Parent).OnValueChanged (EventArgs.Empty);
+			}
+
+			public void OnDoubleClick ()
+			{
+				((Calendar)Parent).OnDoubleClick (EventArgs.Empty);
+			}
+		}
+
+		public Calendar ()
+		{
+			MinDate = DateTime.MinValue;
+			MaxDate = DateTime.MaxValue;
+			Date = DateTime.Now;
+		}
+
+		ICalendarBackend Backend {
+			get { return (ICalendarBackend)BackendHost.Backend; }
+		}
+
+		protected override BackendHost CreateBackendHost ()
+		{
+			return new WidgetBackendHost ();
+		}
+
+		public DateTime Date {
+			get {
+				return Backend.Date;
+			}
+			set {
+				Backend.Date = value;
+			}
+		}
+
+		public DateTime MinDate {
+			get {
+				return Backend.MinDate;
+			}
+			set {
+				Backend.MinDate = value;
+				if (MinDate > MaxDate)
+					MaxDate = MinDate;
+				if (Date < MinDate)
+					Date = MinDate;
+			}
+		}
+
+		public DateTime MaxDate {
+			get {
+				return Backend.MaxDate;
+			}
+			set {
+				Backend.MaxDate = value;
+				if (MaxDate < MinDate)
+					MinDate = MaxDate;
+				if (Date > MaxDate)
+					Date = MaxDate;
+			}
+		}
+
+		[DefaultValue (false)]
+		public bool NoMonthChange {
+			get {
+				return Backend.NoMonthChange;
+			}
+			set {
+				Backend.NoMonthChange = value;
+			}
+		}
+
+		protected virtual void OnValueChanged (EventArgs e)
+		{
+			if (valueChanged != null)
+				valueChanged (this, e);
+		}
+
+		public event EventHandler ValueChanged {
+			add {
+				BackendHost.OnBeforeEventAdd (CalendarEvent.ValueChanged, valueChanged);
+				valueChanged += value;
+			}
+			remove {
+				valueChanged -= value;
+				BackendHost.OnAfterEventRemove (CalendarEvent.ValueChanged, valueChanged);
+			}
+		}
+
+		protected virtual void OnDoubleClick (EventArgs e)
+		{
+			if (doubleClick != null)
+				doubleClick (this, e);
+		}
+
+		public event EventHandler DoubleClick {
```

Comment:
There should be no DoubleClick event, since there is already a ButtonPressed event which can and should be used to handle click events (MultiplePress event args property is meant for detecting e.g. double clicks). So an event that represents an "action" should work like it is done in e.g. TextEntry: Activated. The expected Activated event is a Date selection (with a double click, or by moving the selection cursor and pressing the enter key).